specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json

```diff
@@ -1043,6 +1043,40 @@
         }
       }
     },
+    "WorkspaceRepositoryConfiguration": {
+      "description": "Git integration settings",
+      "type": "object",
+      "properties": {
+        "accountName": {
+          "description": "Account name",
+          "type": "string"
+        },
+        "collaborationBranch": {
+          "description": "Collaboration branch",
+          "type": "string"
+        },
+        "hostName": {
+          "description": "GitHub Enterprise host name. For example: https://github.mydomain.com",
+          "type": "string"
+        },
+        "projectName": {
+          "description": "VSTS project name",
+          "type": "string"
+        },
+        "repositoryName": {
+          "description": "Repository name",
+          "type": "string"
+        },
+        "rootFolder": {
+          "description": "Root folder to use in the repository",
+          "type": "string"
+        },
+        "type": {
```

Comment:
would this be better modeled as x-ms-enum? In the examples I feel workspace and configuration are unnecessary so it could just be VSTS, Github

Reply:
Two things here: 
1) enum I agree with you, but we already check-in this changes, we don't have new deployment/testing windows to make this change now. Will add it as TODO in next release.
2) This naming(VSTS, Github) we don't have control as per ADF documentation we need them as same pattern, please refer to below doc
https://docs.microsoft.com/en-us/rest/api/datafactory/factories/configurefactoryrepo

Example from above doc.
"type": "FactoryVSTSConfiguration",